lib/pam/pam.go

```diff
@@ -280,6 +283,20 @@ func Open(config *Config) (*PAM, error) {
 		return nil, p.codeToError(p.retval)
 	}
 
+	for k, v := range config.Env {
+		// Set a regular OS env var on this process which should be available
+		// to child PAM processes.
+		os.Setenv(k, v)
+
+		// Also set it via PAM-specific pam_putenv, which is respected by
+		// pam_exec (and possibly others), where parent env vars are not.
+		kv := C.CString(fmt.Sprintf("%s=%s", k, v))
+		retval := C._pam_putenv(pamHandle, p.pamh, kv)
```

Comment:
Put `pam_putenv` have any maximum size? If so you might want to cap the length here.

Reply:
http://www.linux-pam.org/Linux-PAM-html/adg-interface-by-app-expected.html#adg-pam_putenv doesn't mention any limits and their source code seems to allocate as much as needed.